pkg/transport/generic_transport_controller.go

```diff
@@ -308,6 +337,7 @@ func (c *genericTransportController) syncHandler(ctx context.Context, objectName
 	}
 
 	if isObjectBeingDeleted(binding) {
+		c.setBindingCares(binding.Name, nil)
```

Comment:
> we shouldn't delete this entry before the next line succeeds (deleteWrappedObjectsAndFinalizer)

At this point it is known that the Binding object is being deleted. Even if the following line returns an error, there is no longer a need to respond to a change in a matching inventory object by enqueuing a reference to the Binding object.

> in addition, it's much more readable to have delete(c.bindingCares, bindingName) rather than calling set function with nil (which internally deletes the entry).

That seems strange to me. All the other manipulations of c.bindingCares are through funcs. It seems nautral to me for this one to also be through a func. It is like there is a little internal abstraction for this data structure about this relation.

Reply:
1. you're correct.
2. I agree with the explanation.